yurt/yurt_core/add.py

```diff
@@ -267,5 +259,147 @@ def role(**kwargs):
+        ".".join((
+            '{}',
+            "yml"
+        ))
+    )
+    return orchestration_path, app_main_path, env_yml_path_template
+
+
+def serialize_vars_from_kwargs_to_python_native(kwargs):
+    if kwargs.get("vars") is not None:
+        try:
+            kwargs["vars"] = json.loads(kwargs.get("vars"))
+        except ValueError:
+            raise ValueError("`--vars` value did not evaluate to proper JSON")
+
+
+# FILE EDITING FUNCTIONS -- Helpers
+
+
+def add_env_var_values_to_env_vars_yaml(path, settings):
+    true_path = path.format(settings.get("env"))
+    add_key_value_to_yaml(true_path, settings.get("vars"))
+    return true_path
+
+
+def add_env_var_values_to_app_main_yaml(path, settings):
+    django_environment = get_value_from_yaml(path, 'django_environment')
+    variables = settings.get("vars")
+
+    for variable, _ in get_iter(variables):
+        django_environment[variable.upper()] = "{{{{ {}|default(omit) }}}}".format(variable)
+
+    remove_value_from_yaml(path, "django_environment")
+    add_key_value_to_yaml(path, {"django_environment": django_environment})
+    return path
+
+
+def add_env_var_values_to_python_settings(path, settings):
+    env_yaml_path = path.format(settings.get("env"))
+    environment_settings_py_file = "/".join(
+        get_value_from_yaml(
+            env_yaml_path,
+            "django_settings_file",
+            "config.settings.base"
+        ).split('.')
+    ) + ".py"
+
+    django_file_path = os.path.join(
+        find_project_folder(),
+        environment_settings_py_file
+    )
+
+    string_to_add = "\n".join([
+        "{0} = os.getenv(\"{0}\")".format(key.upper()) for key in list(settings.get("vars", {}).keys())
+    ])
+    with open(django_file_path, "a") as settings_file:
+        settings_file.write(string_to_add)
+    return django_file_path
+
+
+@add.command()
+@click.option("--env", default=None, help="Name of environment")
+@click.option("--vars", default=None, help="JSON of desired env vars to be added")
+def env_vars(**kwargs):
+    """
+    Add environment variables to specific environments (EXPERIMENTAL)
+    """
+
+    # Get some paths
+    orchestration_path, app_main_path, env_yaml_path_template = get_relevant_paths()
+
+    # Compile a string that prints out the available environments with their option numbers
+    # along with a Dict representing these options
+    envs_available_string, envs_available_options = get_env_question_options_tuple(orchestration_path)
+
+    # Create the question flow
+    env_var_mapping = get_question_flow_mapping(envs_available_string)
+
+    # Make "--vars" option serialized to Python-native Dict
+    serialize_vars_from_kwargs_to_python_native(kwargs)
+
+    # Go over the questions to generate a Dict of settings
+    settings = go_over_questions(
+        env_var_mapping,
+        kwargs,
+        defaults=CallDict({
+            'vars': DeferredCallable(generate_dict_of_things, "environment variables")
```

Comment:
Not a huge fan of the name `generate_dict_of_things` although I'm also a bit stumped on a better name 🤔